projects/components/src/datagrid/datagrid.component.spec.ts

```diff
@@ -73,276 +72,293 @@ describe('DatagridComponent', () => {
                 const firstCall = ['firstRowA', 'secondRowA'];
                 const secondCall = ['firstRowB', 'secondRowB'];
-
-                this.finder.hostComponent.clrDatarowCssClassGetter = (rec: MockRecord, index: number) => {
+                this.hostComponent.clrDatarowCssClassGetter = (rec: MockRecord, index: number) => {
                     return firstCall[index];
                 };
                 this.finder.detectChanges();
-                expect(this.clrGridWidget.getRowsCssClass(0)).toContain(
+                expect(this.clrGridWidget.getRow(0).classes()).toContain(
                     'firstRowA',
                     'Expected the initial class to display for the first row.'
                 );
-                expect(this.clrGridWidget.getRowsCssClass(1)).toContain(
+                expect(this.clrGridWidget.getRow(1).classes()).toContain(
                     'secondRowA',
                     'Expected some different initial class to display for the second row.'
                 );
-
-                this.finder.hostComponent.clrDatarowCssClassGetter = (rec: MockRecord, index: number) => {
+                this.hostComponent.clrDatarowCssClassGetter = (rec: MockRecord, index: number) => {
                     return secondCall[index];
                 };
                 this.finder.detectChanges();
-                expect(this.clrGridWidget.getRowsCssClass(0)).toContain(
+                expect(this.clrGridWidget.getRow(0).classes()).toContain(
                     'firstRowB',
                     'Expected a new class to display for the first row.'
                 );
-                expect(this.clrGridWidget.getRowsCssClass(1)).toContain(
+                expect(this.clrGridWidget.getRow(1).classes()).toContain(
                     'secondRowB',
                     'Expected a different new class to display for the second row.'
                 );
             });
 
             describe('@Input() columns.disableCliptext', () => {
                 beforeEach(function (this: HasFinderAndGrid): void {
-                    this.finder.hostComponent.gridData = {
+                    this.hostComponent.gridData = {
                         items: mockData,
                         totalItems: 2,
                     };
                     this.finder.detectChanges();
                 });
 
                 it('clips text when disableCliptext is unset', function (this: HasFinderAndGrid): void {
-                    this.finder.hostComponent.columns = [{ displayName: 'Name', renderer: 'name' }];
+                    this.hostComponent.columns = [{ displayName: 'Name', renderer: 'name' }];
                     this.finder.detectChanges();
-                    expect(this.clrGridWidget.columnClippedTextDirective(0).disabled).toBeFalsy();
+                    const res = this.clrGridWidget.getCell(0, 0).elements[0].injector.get(ShowClippedTextDirective);
```

Comment:
Instead of asserting this, can we assert that `text-overflow: ellipsis` is set on the style of the cell? When the directive is enabled it sets this property

Reply:
Setting text overflow is the job of the directive, we just need to make sure the directive is disabled

Reply:
Having said this, `his.clrGridWidget.getCell(0, 0).elements[0].injector.get(ShowClippedTextDirective);` looks very hackish, let's talk about how to avoid this.